instrumentation/spring-rabbit/src/main/java/brave/spring/rabbit/SpringRabbitTracing.java

```diff
@@ -53,102 +51,98 @@ public SpringRabbitTracing build() {
 
   final TracingMessagePostProcessor tracingMessagePostProcessor;
   final TracingRabbitListenerAdvice tracingRabbitListenerAdvice;
+  final Field beforePublishPostProcessorsField;
 
   SpringRabbitTracing(Builder builder) {
     Tracing tracing = builder.tracing;
     String remoteServiceName = builder.remoteServiceName;
     this.tracingMessagePostProcessor = new TracingMessagePostProcessor(tracing, remoteServiceName);
     this.tracingRabbitListenerAdvice = new TracingRabbitListenerAdvice(tracing, remoteServiceName);
+    Field beforePublishPostProcessorsField = null;
+    try {
+      beforePublishPostProcessorsField =
+          RabbitTemplate.class.getDeclaredField("beforePublishPostProcessors");
+      beforePublishPostProcessorsField.setAccessible(true);
+    } catch (NoSuchFieldException e) {
+    }
+    this.beforePublishPostProcessorsField = beforePublishPostProcessorsField;
   }
 
-  /** Creates an instrumented rabbit template. */
+  /** Creates an instrumented {@linkplain RabbitTemplate} */
   public RabbitTemplate newRabbitTemplate(ConnectionFactory connectionFactory) {
     RabbitTemplate rabbitTemplate = new RabbitTemplate(connectionFactory);
     rabbitTemplate.setBeforePublishPostProcessors(tracingMessagePostProcessor);
     return rabbitTemplate;
   }
 
-  /** Instruments an existing rabbit template. */
+  /** Instruments an existing {@linkplain RabbitTemplate} */
   public RabbitTemplate decorateRabbitTemplate(RabbitTemplate rabbitTemplate) {
+    // Skip out if we can't read the field for the existing post processors
+    if (beforePublishPostProcessorsField == null) return rabbitTemplate;
```

Comment:
why not just set the value of the field with the processor?

Reply:
if this is null, we had a reflection error which I'm not assuming will get better :)